Fatras/include/ActsFatras/Kernel/detail/SimulationActor.hpp

```diff
@@ -167,7 +167,8 @@ struct SimulationActor {
         // again: interact only if there is valid material to interact with
         if (slab) {
           // adapt material for non-zero incidence
-          auto normal = surface.normal(state.geoContext, local);
+          auto normal = surface.normal(state.geoContext, before.position(),
+                                       before.direction());
```

Comment:
I guess a material surface should always be a regular one right? just querying with the local position should be fine then

Reply:
In principle I'd agree, but then I'd have to `dynamic_cast` and check here if it's `RegularSurface`. Maybe that's a better idea?